File: sdk/python/src/tinyverse/api/stats.py
from __future__ import annotations

from typing import Any

from ..http import HttpClient


class StatsApi:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    async def overview(self) -> Any:
        return await self._http.get("/stats")

    async def agents(self) -> Any:
        return await self._http.get("/stats/agents")

    async def transactions(self) -> Any:
        return await self._http.get("/stats/transactions")

    async def volume(self) -> Any:
        return await self._http.get("/stats/volume")

    async def fees(self) -> Any:
        return await self._http.get("/stats/fees")

File: sdk/rust/src/api/stats.rs
use std::sync::Arc;

use crate::error::Result;
use crate::http::HttpClient;
use crate::types::{AgentStats, FeeStats, StatsSnapshot, TransactionStats, VolumeStats};

pub struct StatsApi {
    http: Arc<HttpClient>,
}

impl StatsApi {
    pub fn new(http: Arc<HttpClient>) -> Self {
        Self { http }
    }

    pub async fn overview(&self) -> Result<StatsSnapshot> {
        self.http.get("/stats", None).await
    }

    pub async fn agents(&self) -> Result<AgentStats> {
        self.http.get("/stats/agents", None).await
    }

    pub async fn transactions(&self) -> Result<TransactionStats> {
        self.http.get("/stats/transactions", None).await
    }

    pub async fn volume(&self) -> Result<VolumeStats> {
        self.http.get("/stats/volume", None).await
    }

    pub async fn fees(&self) -> Result<FeeStats> {
        self.http.get("/stats/fees", None).await
    }
}
